server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -337,17 +343,23 @@ protected ConstraintSecurityHandler configureSpnego(Server server,
       AvaticaServerConfiguration config) {
     final String realm = Objects.requireNonNull(config.getKerberosRealm());
     final String principal = Objects.requireNonNull(config.getKerberosPrincipal());
-
-    // A customization of SpnegoLoginService to explicitly set the server's principal, otherwise
-    // we would have to require a custom file to set the server's principal.
-    PropertyBasedSpnegoLoginService spnegoLoginService =
-        new PropertyBasedSpnegoLoginService(realm, principal);
+    final String principalWithoutRealm = principal.split("@", 2)[0];
+    final String[] principalParts = principalWithoutRealm.split("/", 2);
+    final String serviceName = principalParts[0];
+    final String hostname = principalParts[1];
```

Comment:
Currently Avatica server is configured with a principal. The Jetty implementation builds a principal from the service name (ie: `avatica`) and the hostname (ie: `localhost`). So breaking apart here to get into necessary parts.